docs/contributor_tips.md

```diff
@@ -0,0 +1,58 @@
+# Maintainer Tips
+
+This is an opinionated guide documenting workflows that some members of the team
+have found useful. It is focused on meta-tooling, not on IREE code specifically
+(you will find the latter in the
+[Developer Overview](../developer_overview.md)) It is certainly possible to use
+workflows other than these, but some common tasks, especially for maintainers
+will likely be made easier if you use these flows. It assumes a basic knowledge
+of `git` and GitHub and suggests some specific ways of using it.
+
+## Git Structure
+
+Work primarily from a clone of the repository on your development machine. Any
+local branches named the same as persistent branches from the
+[main repository](https://github.com/google/iree) (currently `main`, `google`,
+and `stable`) are pristine (though potentially stale) copies. You only
+fastforward these to match upstream and otherwise do development on other
+branches. When sending PRs, you push to a different branch on your fork and
+create the PR from there.
+
+### Setup
+
+1. Create a fork of the main repository.
+2. Clone from this repository or the main repository.
+3. Rename your remotes to `upstream` and `fork`, so that it's abundantly clear
```

Comment:
I just use `origin` and `${USERNAME}` (matching the actual name of the fork on GitHub). Much clearer to me :P

When I pull down commits from other users' forks, there's also no confusion about which remote I'm working with.

Reply:
Yeah I'm happy to use someone else's convention here. I just wrote down what I use. I like the idea of just naming each remote after its owner. It scales nicely. But it would make scripting very difficult: Determining someone's GitHub username from the CLI would require interfacing with the GitHub API I think (there isn't a nice `gh` CLI command for it). I'm also wary of calling the upstream remote `origin`. Lots of things will assume origin for pushing and such, and especially with admin privileges it's pretty dangerous.

Looking at some conventions on this. I think it's actually more standard to call your fork `origin` and the main repo `upstream`:
- https://stackoverflow.com/q/38965156
- https://github.blog/2015-07-29-git-2-5-including-multiple-worktrees-and-triangular-workflows/
- https://www.atlassian.com/git/tutorials/comparing-workflows/forking-workflow

I got confused enough with the remote naming that I just bailed and decided to call nothing `origin` since then I have to think about it every time, but I think this has caused my local git repo some confusion because it doesn't automatically set up new branches to push to my fork and I have to be explicit about the remote each time.

Reply:
Yes, `origin` and `upstream` is was, following https://docs.github.com/en/github/using-git/pushing-commits-to-a-remote-repository#remotes-and-forks. If you don't push directly to `upstream`, that's actually okay. Might be more confusing if you also directly contribute to `upstream`.

Reply:
`origin` is the default name that people will get following our getting started guides when they `git clone`. Having tips mention renaming to `upstream` and setting a fork name that is clear SGTM.

Reply:
Ok let's go with calling the upstream `upstream` and your fork `origin`, since that seems to be standard terminology. I'll update that here and then update the scripts to use that as the default as well.

Reply:
Updated to that. I think I'm going to keep my "fork" alias and just also add "origin". That way it defaults to the right thing but also I can type fork and it knows what I'm talking about.